src/mon/OSDMonitor.cc

```diff
@@ -5897,7 +5897,7 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
     }
 
     err = newcrush.adjust_subtree_weightf(g_ceph_context, id, w);
-    if (err < 0)
+    if (err <= 0)
```

Comment:
This might not be right.  If the pending_crush isn't committed yet, even though it is a no-op, we can't reply because it isn't committed yet.